animate/adapt.py

```diff
@@ -111,27 +138,84 @@ def interpolate(self, f):
         Interpolate a :class:`.Function` into the corresponding :class:`.FunctionSpace`
         defined on the adapted mesh.
 
-        :param: the scalar :class:`.Function` on the initial mesh
-        :return: its interpolation onto the adapted mesh
+        :arg f: a Function on the initial mesh
+        :type f: :class:`firedrake.function.Function`
+        :returns: its interpolation onto the adapted mesh
+        :rtype: :class:`firedrake.function.Function`
         """
         raise NotImplementedError(
             "Consistent interpolation has not yet been implemented in parallel"
         )  # TODO
 
 
-def adapt(mesh, *metrics, name=None):
+def adapt(mesh, *metrics, name=None, serialise=False, remove_checkpoints=True):
```

Comment:
That's not true. If we are running 3D in parallel passing `serialise=True` then it will stay true.

Reply:
I think there is confusion with the terminology used, which might be the cause of the confusion in https://github.com/pyroteus/animate/pull/72#discussion_r1517656191 too :) why would we pass `serialise=True` if we are running 3D in parallel? In that case, we do NOT serialise since it can work in parallel, right?

Reply:
There's no point serialising if you're already running in serial - as you've noted, it will be more expensive to use the checkpointing approach.

The 3D parallel case with ParMmg is not very well tested and likely has bugs, so I can see cases where it would be useful to serialise and run with Mmg.

Reply:
We might want to serialise 3D parallel because ParMmg can be somewhat unreliable.

Reply:
I still find this very confusing :) So if we know that parallel adaptation only works in 3D, I don't like having the `serialise` argument to `adapt`. Especially since we have the `if` checks in place. 

So if the idea is to adapt in parallel only if it's 3D unless we specifically want to serialise 3D due to ParMmg being unreliable sometimes, how about this:

```
def adapt(mesh, *metrics, name=None, serialise3d=False, remove_checkpoints=True):
    nprocs = COMM_WORLD.size

    # Parallel adaptation is currently only supported in 3D
    if mesh.topological_dimension() != 3:
        serialise = nprocs >= 1  # always serialise if it is not 3D
    elif serialise3d:
        serialise = nprocs > 1  #  serialise 3D if serialise3d AND if we aren't already running in serial (i.e. avoid unnecessary checkpointing)
```

Reply:
Could also raise a warning if `serialise=True`.